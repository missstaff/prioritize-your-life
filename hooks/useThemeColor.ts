import { useColorScheme } from "react-native";
import { COLORTHEME } from "@/constants/Colors";


export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof COLORTHEME.light & keyof typeof COLORTHEME.dark
) {
  const theme = useColorScheme() ?? "light";
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return COLORTHEME[theme][colorName];
  }
}
