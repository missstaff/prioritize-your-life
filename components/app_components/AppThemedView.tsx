import { View, type ViewProps } from "react-native";
import { useThemeColor } from "@/hooks/useThemeColor";
import { COLORTHEME } from "@/constants/Colors";

export type ThemedViewProps = ViewProps & {
  darkColor?: string;
  lightColor?: string;
};


/**
 * Renders a themed view component with customizable style and other props.
 * @param {ThemedViewProps} props - The props for the AppThemedView component.
 * @param {ViewStyle} props.style - The style object for the component.
 * @param {any} props.otherProps - Other props to be spread on the component.
 * @returns {JSX.Element} The rendered AppThemedView component.
 */
export function AppThemedView({ style, ...otherProps }: ThemedViewProps):JSX.Element {
  const backgroundColor = useThemeColor(
    { light: COLORTHEME.light.background, dark: COLORTHEME.dark.background },
    "background"
  );

  return <View style={[{ backgroundColor }, style]} {...otherProps} />;
}
