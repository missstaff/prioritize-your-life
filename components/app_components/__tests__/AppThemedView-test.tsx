import * as React from "react";
import renderer from "react-test-renderer";
import { AppThemedView } from "../AppThemedView";
import { useThemeColor } from "@/hooks/useThemeColor";
import { View } from "react-native";

describe("AppThemedView Tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with default theme", () => {
    const tree = renderer.create(<AppThemedView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with dark theme", () => {
    const mockDarkColor = "#000000";
    (useThemeColor as jest.Mock).mockReturnValue(mockDarkColor);

    const tree = renderer.create(<AppThemedView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with light theme", () => {
    const mockLightColor = "#ffffff";
    (useThemeColor as jest.Mock).mockReturnValue(mockLightColor);

    const tree = renderer.create(<AppThemedView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies correct background color from theme", () => {
    const mockColor = "#ff6347";
    (useThemeColor as jest.Mock).mockReturnValue(mockColor);

    const component = renderer.create(<AppThemedView />);
    const view = component.root.findByType(View);

    expect(view.props.style).toContainEqual({ backgroundColor: mockColor });
  });
});
