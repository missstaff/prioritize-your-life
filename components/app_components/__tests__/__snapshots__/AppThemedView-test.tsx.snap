// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppThemedView Tests renders correctly with dark theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000000",
      },
      undefined,
    ]
  }
/>
`;

exports[`AppThemedView Tests renders correctly with default theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      undefined,
    ]
  }
/>
`;

exports[`AppThemedView Tests renders correctly with light theme 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
/>
`;
